bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/EntsoeHandler.java

```diff
@@ -0,0 +1,314 @@
+                    config.historicDays);
+            needToFetch = true;
+        }
+
+        if (updateHistoricDaysInitially && historicDaysInitially != config.historicDays) {
+            historicDaysInitially = config.historicDays;
+        }
+
+        return needToFetch;
+    }
+
+    private void refreshPrices() {
+        logger.trace("refreshPrices()");
+        if (!isLinked(EntsoeBindingConstants.CHANNEL_SPOT_PRICES)) {
+            logger.debug("Channel {} is not linked, cant update channel", EntsoeBindingConstants.CHANNEL_SPOT_PRICES);
+            return;
+        }
+
+        config = getConfigAs(EntsoeConfiguration.class);
+
+        ZonedDateTime startUtc = currentUtcTimeWholeHours()
+                .minusDays(needToFetchHistoricDays() ? config.historicDays : 1).withHour(22);
+        ZonedDateTime endUtc = currentUtcTimeWholeHours().plusDays(1).withHour(22);
+
+        boolean needsUpdate = lastDayAheadReceived.equals(ZonedDateTime.of(LocalDateTime.MIN, ZoneId.of("UTC")))
+                || responseMap == null || needToFetchHistoricDays(true);
+
+        boolean hasNextDayValue = needsUpdate ? false
+                : responseMap.entrySet().stream()
+                        .anyMatch(x -> x.getKey().getDayOfYear() == currentUtcTime().plusDays(1).getDayOfYear());
+        boolean readyForNextDayValue = needsUpdate ? true
+                : currentUtcTime().isAfter(currentUtcTime().withHour(config.spotPricesAvailableUtcHour));
+
+        // Update whole time series
+        if (needsUpdate || (!hasNextDayValue && readyForNextDayValue)) {
+            logger.debug("Updating timeseries");
+            Request request = new Request(config.securityToken, config.area, startUtc, endUtc);
+            Client client = new Client();
+            boolean success = false;
+
+            try {
+                responseMap = client.doGetRequest(request, 30000);
+                TimeSeries baseTimeSeries = new TimeSeries(EntsoeBindingConstants.TIMESERIES_POLICY);
+                for (Map.Entry<ZonedDateTime, Double> entry : responseMap.entrySet()) {
+                    BigDecimal exchangedKwhPrice = getExchangedKwhPrice(entry.getValue());
+                    State baseState = new QuantityType<>(exchangedKwhPrice + " " + baseUnit.getName() + "/kWh");
+                    baseTimeSeries.add(entry.getKey().toInstant(), baseState);
+                }
+                lastDayAheadReceived = currentUtcTime();
+                sendTimeSeries(EntsoeBindingConstants.CHANNEL_SPOT_PRICES, baseTimeSeries);
+                updateCurrentHourState(EntsoeBindingConstants.CHANNEL_SPOT_PRICES);
+                scheduler.schedule(this::triggerChannelSpotPricesReceived, 30, TimeUnit.SECONDS);
+                success = true;
+            } catch (EntsoeResponseException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                logger.error("{}", e.getMessage());
```

Comment:
Log level `error` is not appropriate here, and logging is redundant, since the status update is already logged.
```suggestion
```